lesson1

amzhulanov opened the pull request.

amzhulanov committed:
Метод, который меняет элементы местами

amzhulanov committed:
Метод Main - работа с фруктами
ChangeArray - меняю жлементы массива местами и преобразую массив в список

usharik commented on src/ru/geekbrains/lesson1/homework1/Box.java:
```diff
@@ -0,0 +1,30 @@
+    public boolean compare(Box<?> box){
+        return Math.abs(this.getWeight()-box.getWeight())<0.0001;
+    }
+
```
Нужен еще метод для пересыпания фруктов из другой коробки.

amzhulanov replied:
Упустил, исправлюсь

usharik commented on src/ru/geekbrains/lesson1/homework1/ChangeArray.java:
```diff
@@ -0,0 +1,36 @@
+        item=numListOut[xPos];
+        numListOut[xPos]=numListOut[yPos];
+        numListOut[yPos]=item;
+        List<T> numListIn=new ArrayList<>();
```
Не совсем понимаю, в чем смысл этих операций. Если вы хотели оставить исходный массив неизменным, то его нужно было скопировать перед обменой элементов местами.

amzhulanov replied:
item=numListOut[xPos];
        numListOut[xPos]=numListOut[yPos];
        numListOut[yPos]=item;
В первых 3х строчках я меняю элементы массива местами
В следующих двух строчках происходит преобразование массива в список

amzhulanov committed:
Перенёс исполнение методов из Main в Box. Класс Main  убил.
Добавил метод пересыпания фруктов.
В ChangeArray добавил комментарии, поясняющие, где выполнено задание №1 и где задание №2
